src/globus_sdk/experimental/tokenstorage/json.py

```diff
@@ -40,8 +41,15 @@ def __init__(self, filename: pathlib.Path | str, *, namespace: str = "DEFAULT"):
         :param namespace: A user-supplied namespace for partitioning token data
         """
         self.filename = str(filename)
+        self._ensure_containing_dir_exists()
```

Comment:
If we're adding this, shouldn't we do it on the parent class, not here?

Reply:
Sure, that's a fair question. Would you like to open that PR or me?

Reply:
Sure, I'll work that up!

(Also, no worries about merge/review order. I noticed it, and we'll sort out now. :smile: )